src/store/generator.ts

```diff
@@ -0,0 +1,164 @@
+import PdfWorker from '../Utility/pdf.worker.ts';
+import { rethrow } from '../Utility/errors';
+import { UserRequests } from '../DataType/UserRequests';
+
+export interface GeneratorSpecificState {
+    ready: boolean;
+    download_active: boolean;
+    download_url?: string;
+    download_filename?: string;
+    fillFields: IdDataElement[];
+    fillSignature: Signature;
+    pdfWorker?: PdfWorker;
+    setBusy: () => void;
+    setReady: () => void;
+    setDownload: (download_active: boolean, download_url?: string, download_filename?: string) => void;
+    refreshFillFields: () => void;
+    initiatePdfGeneration: () => void;
+    renderLetter: () => void;
+    resetInitialConditions: (response_to?: string, response_type?: ResponseType) => Promise<void>;
+}
+
+export type GeneratorState = RequestState<Request> & CompanyState & GeneratorSpecificState;
+
+export const {
+    Provider: RequestGeneratorProvider,
+    useStore: useGeneratorStore,
+    useStoreApi: useGeneratorStoreApi,
+} = createContext<GeneratorState>();
+
+const savedIdData = new SavedIdData();
+
+const createGeneratorSpecificStore: StoreSlice<GeneratorSpecificState, RequestState<Request> & CompanyState> = (
+    set,
+    get
+) => ({
+    ready: true,
+    download_active: false,
+    fillFields: [],
+    fillSignature: { type: 'text', name: '' },
+    setReady: () => {
+        set({ ready: true });
+    },
+    setBusy: () => {
+        set({ ready: false });
+    },
+    setDownload: (download_active, download_url, download_filename) =>
+        set({
+            download_active,
+            download_url,
+            download_filename,
+        }),
+    refreshFillFields: () => {
+        if (Privacy.isAllowed(PRIVACY_ACTIONS.SAVE_ID_DATA)) {
+            savedIdData.getAll(false).then((fillFields) => fillFields && set({ fillFields }));
+            savedIdData.getSignature().then((fillSignature) => fillSignature && set({ fillSignature }));
+        }
+    },
+    initiatePdfGeneration: () => {
+        let pdfWorker = get().pdfWorker;
+        if (!pdfWorker) pdfWorker = new PdfWorker();
+
+        if ((window as typeof window & { hugoDevMode: boolean }).hugoDevMode) {
+            // copy the worker to window if we are in a dev env to enable easy testing
+            (window as typeof window & { pdfWorker: PdfWorker }).pdfWorker = pdfWorker;
+        }
+        const onMessage = (message: MessageEvent<{ blob_url: string; filename: string }>) => {
+            get().setDownload(true, message.data.blob_url, message.data.filename);
+            get().setReady();
+        };
+        const onError = (error: ErrorEvent) => rethrow(error, 'PdfWorker error');
+
+        pdfWorker.addEventListener('message', onMessage);
+        pdfWorker.addEventListener('error', onError);
+
+        set({ pdfWorker });
+
+        return () => {
+            pdfWorker!.removeEventListener('message', onMessage);
+            pdfWorker!.removeEventListener('error', onError);
```

Comment:
I know that we currently don't ever get rid of the worker. But why not be safe here?

```suggestion
            pdfWorker?.removeEventListener('message', onMessage);
            pdfWorker?.removeEventListener('error', onError);
```